fapolicy_analyzer/ui/policy_rules_admin_page.py

```diff
@@ -500,6 +525,40 @@ def on_userTabs_switch_page(self, notebook, page, page_num):
     def on_refresh_clicked(self, *args):
         self.__refresh()
 
+    def on_timeSelectBtn_clicked(self, *args):
+        def plural(count):
+            return "s" if count > 1 else ""
+
+        time_dialog = TimeSelectDialog()
+        time_dialog.get_object("timeComboBox").set_active_id(self.time_unit)
+        buff = time_dialog.get_object("timeEntryField").get_buffer()
+        buff.set_text(str(self.time_number), len(str(self.time_number)))
+        resp = time_dialog.get_ref().run()
+        time_dialog.get_ref().hide()
+        time_unit = time_dialog.get_object("timeComboBox").get_active_id()
+        time_number = int(time_dialog.get_object("timeEntryField").get_buffer().get_text())
```

Comment:
You could simplify a lot of this code in here if the `TimeSelectDialog` implemented getters and setters for `active_id` and `text` that hide all the implementation details.